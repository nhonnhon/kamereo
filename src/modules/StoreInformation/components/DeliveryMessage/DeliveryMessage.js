import React from 'react'
import { Input, Button, Checkbox } from 'components'
import { Lang } from 'constants/language'

export default class DeliveryMessage extends React.Component {
  state = {
    deliveryMessageId: [
      {
        text: 'Lorem ipsum is dolor assets',
        checked: true,
      },
      {
        text: '',
        checked: false,
      },
      {
        text: '',
        checked: false,
      },
      {
        text: '',
        checked: false,
      },
      {
        text: '',
        checked: false,
      },
    ],
  }

  onChangeCheckbox = (index, checked) => {
    const { deliveryMessageId } = this.state
    deliveryMessageId[index].checked = !checked

    this.setState({
      deliveryMessageId,
    })
  }

  onChangeInput = (index, value) => {
    const { deliveryMessageId } = this.state
    deliveryMessageId[index].text = value

    this.setState({
      deliveryMessageId,
    })
  }

  renderInputMessage = ({ text, checked }, index) => (
    <div className="d-flex mt-1" key={index}>
      <Checkbox className="mr-1" checked={checked} onCheck={() => this.onChangeCheckbox(index, checked)} />
      <Input
        id={index}
        value={text}
        type="text"
        className="flex1"
        onChangeId={this.onChangeInput}
        placeholder={Lang.writeYourMessage}
      />
    </div>
  )

  render() {
    const { deliveryMessageId } = this.state

    return (
      <div className="panel delivery-message">
        <p className="text black uppercase">{Lang.deliveryMessage}</p>
        {deliveryMessageId.map((item, index) => this.renderInputMessage(item, index))}
        <Button
          size="small"
          color="gray1"
          className="mt-2"
          onClick={() => {}}
          textColor="black"
          label={Lang.update}
        />
      </div>
    )
  }
}
