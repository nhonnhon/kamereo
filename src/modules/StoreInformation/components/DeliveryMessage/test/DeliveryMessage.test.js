import React from 'react'
import { create } from 'react-test-renderer'
import DeliveryMessage from '../DeliveryMessage'

describe('DeliveryMessage component', () => {
  test('Matches the snapshot', () => {
    const deliveryMessage = create(<DeliveryMessage />)
    expect(deliveryMessage.toJSON()).toMatchSnapshot()
  })
})
