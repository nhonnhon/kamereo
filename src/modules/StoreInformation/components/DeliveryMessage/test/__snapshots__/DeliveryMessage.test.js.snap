// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryMessage component Matches the snapshot 1`] = `
<div
  className="panel delivery-message"
>
  <p
    className="text black uppercase"
  >
    Delivery default message
  </p>
  <div
    className="d-flex mt-1"
  >
    <div
      className="checkbox checked mr-1"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "",
        }
      }
    >
      <div
        className="icon"
      />
      <input
        className="hide"
        type="checkbox"
      />
      <label
        className="pad-left "
      >
        <div
          className="content"
        />
      </label>
    </div>
    <div
      className="sc-input-container flex1 no-label"
    >
      <div
        className="d-flex align-left flex1"
      >
        <div
          className="input-container "
        >
          <input
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            placeholder="Write your message"
            type="text"
            value="Lorem ipsum is dolor assets"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="d-flex mt-1"
  >
    <div
      className="checkbox  mr-1"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "",
        }
      }
    >
      <div
        className="icon"
      />
      <input
        className="hide"
        type="checkbox"
      />
      <label
        className="pad-left "
      >
        <div
          className="content"
        />
      </label>
    </div>
    <div
      className="sc-input-container flex1 no-label"
    >
      <div
        className="d-flex align-left flex1"
      >
        <div
          className="input-container "
        >
          <input
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            placeholder="Write your message"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="d-flex mt-1"
  >
    <div
      className="checkbox  mr-1"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "",
        }
      }
    >
      <div
        className="icon"
      />
      <input
        className="hide"
        type="checkbox"
      />
      <label
        className="pad-left "
      >
        <div
          className="content"
        />
      </label>
    </div>
    <div
      className="sc-input-container flex1 no-label"
    >
      <div
        className="d-flex align-left flex1"
      >
        <div
          className="input-container "
        >
          <input
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            placeholder="Write your message"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="d-flex mt-1"
  >
    <div
      className="checkbox  mr-1"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "",
        }
      }
    >
      <div
        className="icon"
      />
      <input
        className="hide"
        type="checkbox"
      />
      <label
        className="pad-left "
      >
        <div
          className="content"
        />
      </label>
    </div>
    <div
      className="sc-input-container flex1 no-label"
    >
      <div
        className="d-flex align-left flex1"
      >
        <div
          className="input-container "
        >
          <input
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            placeholder="Write your message"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="d-flex mt-1"
  >
    <div
      className="checkbox  mr-1"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "",
        }
      }
    >
      <div
        className="icon"
      />
      <input
        className="hide"
        type="checkbox"
      />
      <label
        className="pad-left "
      >
        <div
          className="content"
        />
      </label>
    </div>
    <div
      className="sc-input-container flex1 no-label"
    >
      <div
        className="d-flex align-left flex1"
      >
        <div
          className="input-container "
        >
          <input
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            placeholder="Write your message"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <a
    className="btn  small gray1 mt-2"
    href="javascript: void(0)"
    onClick={[Function]}
  >
    <span
      className="text  black small"
    >
      Update
    </span>
  </a>
</div>
`;
